packages/dappkit/src/index.ts

```diff
@@ -110,17 +103,18 @@ export enum GasCurrency {
   cGLD = 'cGLD',
 }
 
-async function getGasCurrencyContract(
+async function getGasCurrencyContractAddress(
   web3: Web3,
   gasCurrency: GasCurrency
-): Promise<CeloTokenType> {
+): Promise<string> {
+  const kit = newKitFromWeb3(web3)
```

Comment:
Would it be possible/reasonable to have an instance of `ContractKit` persist, rather than creating a new instance each time it's used? `ContractKit` avoids extra contract calls by caching things like contract addresses, hitting the Registry contract only on the first call to `addressFor`. But, reinstantiating the kit for each method call doesn't take advantage of that.

Reply:
I think that is definitely a real issue. I didn't want to prescribe the use of ContractKit for DappKit users, but maybe we should given that we use it under the hood

Reply:
I agree with @yerdua here. There are a couple of things that `ContractKit` needs to do:

* fetch the registry address
* in the future, check compatibility

i believe both things can be expensive. At the same time, we are doing those calls to a full node, that won't be happy if we do calls that we don't need

Reply:
I agree with you that in principle that we should reuse kit instances.

I do wonder whether we really need to expose DAppKit developers to ContractKit as that increases the cognitive burden. I could see it either way. Anyways, for now, I'll make the change to requiring a kit instance